fontforge/tottf.c

```diff
@@ -3333,7 +3333,7 @@ static void setos2(struct os2 *os2,struct alltabs *at, SplineFont *sf,
 	os2->version = 3;
     if ( sf->use_typo_metrics || sf->weight_width_slope_only )
 	os2->version = 4;
-    if ( sf->os2_version!=0 )
+    if ( sf->os2_version > os2->version )
```

Comment:
I think the correct fix here is for the font info dialog to warn the user if he is setting a field not supported by the selected version, not to override it in this low level code. Otherwise it makes no sense to even allow setting OS/2 version explicitly (not necessarily a bad thing) and the support for it can just be dropped.